### PR TITLE
frontend: Add infinite token approval checkbox

### LINKED ISSUE #1071: Adding a "I want to approve infinite sum" button
# Rationale
We had several of our users complain/point out that it is a little weird our bridge is requesting to have access to their entire balance.
image

Alternatively, a community member has suggested a potential UX solution.
image

# Implementation

### PR DESCRIPTION
**Pull request by manuelwedler**
closes #1071 

Adds a checkbox below the transfer dialog which allows users to decide to approve an infinite amount of their tokens. Makes the token allowance transaction of future transfers obsolete for them.
I did some refactoring of the css of the app's main div, because we were not using tailwind there.

### CODE REVIEW

**Review by GabrielBuragev — CHANGES_REQUESTED** [review]
Very nice feature, thanks for implementing it! 
Code wise, everything looks top notch. 1 nitpick and 1 suggestion.

This solution does not include the option for removing the infinite token allowance right? 
I think it would be nice to give the user the option to revoke his allowance at a later point.
I was thinking how we could utilize the current pattern you introduced to add that feature as well. He would need to submit another transaction in order to overwrite the allowance value. If it is not too difficult, can you please do it in this PR?

I would also prefer to see a button somewhere on the app for removing his infinite allowance but that should be in its own issue.
This could be done as a part of https://github.com/beamer-bridge/beamer/issues/1194 so i will update the issue description.
I imagine a menu in the header in which the user can see the state of his connection as well as disconnect & revoke infinite token allowances (if set). Uniswap's UI is a good example. 

Besides this i would like @DBecane to review the texts and see if she has any suggestions:

image

**Comment by DBecane** [comment]
@GabrielBuragev @manuelwedler amazing feature! 
Love it, improves the UX drastically. Here is my suggestion in terms of text to make it easier to read and understand: 

- [ ] Remember my wallet address for future transfers

Infobox when hovering over the '?' : Save time and money on transaction fees by ticking the 'Remember my wallet address' box. This will allow you to skip the access permission step when making future transfers with Beamer. You can always change your preferences by simply unticking the box.

**Comment by manuelwedler** [comment]
About revoking the unlimited allowance:
So the idea would be to revoke the allowance if the user unticks the box later? The transaction would only be triggered by doing another transfer right?

**Comment by manuelwedler** [comment]
We just decided to not do the revocation inside this PR. I will create an issue for it.

@DBecane Therefore, I will adapt your Infobox text by removing the last sentence. Is this fine?

> Infobox: 
> Save time and money on transaction fees by ticking this box. This allows you to grant continuous token access to our contracts, so you can skip the recurring access permission step for future transfers with Beamer.

Also I think the text is not precise. The allowance is always per rollup. If the user does a transfer from a different rollup, they would need to do an approval transaction again. `for future transfers with Beamer.` sounds like they will never have to do it again.

**Comment by DBecane** [comment]
> We just decided to not do the revocation inside this PR. I will create an issue for it.
> 
> @DBecane Therefore, I will adapt your Infobox text by removing the last sentence. Is this fine?
> 
> > Infobox:
> > Save time and money on transaction fees by ticking this box. This allows you to grant continuous token access to our contracts, so you can skip the recurring access permission step for future transfers with Beamer.
> 
> Also I think the text is not precise. The allowance is always per rollup. If the user does a transfer from a different rollup, they would need to do an approval transaction again. `for future transfers with Beamer.` sounds like they will never have to do it again.

I understand, how do other dapps do it?

**Comment by andfletcher** [comment]
@manuelwedler please check how other project phrase the button for inifinit amount. Lets not take the wording "remember my address" since this would be misleading. The message needs to provide the information that the user is providing infinite amount regarding the token approval.

**Comment by istankovic** [comment]
> Lets not take the wording "remember my address" since this would be misleading. The message needs to provide the information that the user is providing infinite amount regarding the token approval.

I totally agree with @andfletcher here, "remember my address" is misleading and does not say what is really going on.

**Comment by andfletcher** [comment]
> Approve maximum token allowance - Save time and money on transaction fees by ticking this box to skip the recurring access permission step for future transfers with Beamer. You only need to approve once per token and rollup/chain. Please keep in mind the risks if you use this option.

@manuelwedler final wording. First part is the text on the website. Second part (separated by the "-" )is the info text.

File: frontend/src/composables/useTokenAllowance.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        token.value,
+        signerAddress.value,
+        sourceChain.value.requestManagerAddress,
+      );
+    } else {
+      allowance.value = undefined;
+    }
+  }
+
+  watch([token, provider, sourceChain, signerAddress], updateTokenAllowance, { immediate: true });
```
[the remark this answers is not in the document]
Comment (manuelwedler) [reply frontend/src/composables/useTokenAllowance.ts:33]: yeah, true should be sufficient

**Comment by manuelwedler** [comment]
Issue for the revocation: #1512

**Review by GabrielBuragev — APPROVED** [review]
Awesome! Thanks for addressing all the things.

**Pull request merged**